pkg/operator/hive/hiveadmission.go

```diff
@@ -93,3 +99,21 @@ func (r *ReconcileHiveConfig) deployHiveAdmission(haLog log.FieldLogger, h *reso
 
 	return nil
 }
+
+func (r *ReconcileHiveConfig) redeployHiveAdmission(haLog log.FieldLogger, instance *hivev1.HiveConfig, recorder events.Recorder) error {
+	daemonSet, err := r.kubeClient.AppsV1().DaemonSets(hiveNamespace).Get(admissionDaemonSetName, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		// Daemonset doesn't exist yet, nothing to do
+		return nil
+	}
+	if err != nil {
+		haLog.WithError(err).Error("cannot retrieve admission daemonset")
+		return err
+	}
+	_, _, err = resourceapply.ApplyDaemonSet(r.kubeClient.AppsV1(), recorder, daemonSet, daemonSet.Generation, true)
```

Comment:
@ShivamChamoli is in the process of converting this to a Deployment, so this will need to be updated for a deployment but IIRC the resourceapply module had similar functions for Deployments. 

I'm not thrilled to see this library back in use after going to apply, but is presumably this is better than deleting pods as Kube will more safely roll the deployment?

Reply:
basically, yes... it sets a random annotation on both the daemonset/deployment and its template